AudiusProject/audius-protocol
# Add replica set on signup

### Description
_What is the purpose of this PR? What is the current behavior? New behavior? Relevant links (e.g. Trello) and/or information pertaining to PR?_

**Purpose:**
To add a replica set to new users on signup. This PR is purely additive.

**Current behavior:**
On signup, new users do not have a replica set. This causes data to be centralized on the User Metadata node. 

**New behavior:**
Assign a replica set to new users so that new user data is stored across different Content Nodes.

### Tests
_List the manual tests and repro instructions to verify that this PR works as anticipated. Include log analysis if possible.\
:exclamation: If this change impacts clients, make sure that you have tested the clients :exclamation:_

Test that:
1. new users on signup get assigned a replica set, and that user actions trigger syncs
2. existing users (creator and non-creator) can use audius as expected and not encounter any issues

Test steps:

md = metadata -> check /users?handle=<handle> and /ipfs/<hash> across replica set
check clock = await audiusLibs.creatorNode.getClockValuesFromReplicaSet() in browser console

Test scenarios:
------- check out master branches of dapp and protocol -------
1. Existing user
   - og non-creator
      - check md/clock
      - update profile
      - check md/clock
   - og creator
      - upload track
      - check md/clock

------- check out `vg-add-replica-set-signup-audbackendchanges` (dapp) and (protocol) `vg-add-replica-set-on-signup-umd` -------

2. Stay logged in as creator
   - upload track
   - check md/clock
   - update profile
   - check md/clock
   - update track
   - check md/clock
   - change rests
   - check md/clock
3. Login as og non-creator
   - update profile 
   - check md/clock
   - do steps in 2
4. Create new users and go through the flow
    - creator
        - steps in 2
    - non-creator
        - steps in 3

logs (there is a lot):
[deprecate_um_logs with Markup.pdf](https://github.com/AudiusProject/audius-protocol/files/5785042/deprecate_um_logs.W.Markup.pdf)
